storage/src/main/java/io/delta/storage/HadoopFileSystemLogStore.java

```diff
@@ -68,6 +70,72 @@ public Path resolvePathOnPhysicalStorage(
         return path.getFileSystem(hadoopConf).makeQualified(path);
     }
 
+    /**
+     * An internal write implementation that uses FileSystem.rename().
+     * <p>
+     * This implementation should only be used for the underlying file systems that support atomic
+     * renames, e.g., Azure is OK but HDFS is not.
+     */
+    protected void writeWithRename(
+            Path path,
+            Iterator<String> actions,
+            Boolean overwrite,
+            Configuration hadoopConf) throws IOException {
+        FileSystem fs = path.getFileSystem(hadoopConf);
+
+        if (!fs.exists(path.getParent())) {
+            throw new FileNotFoundException(
+                    String.format("No such file or directory: %s", path.getParent())
+            );
+        }
+        if (overwrite) {
+            final FSDataOutputStream stream = fs.create(path, true);
+            try {
+                while (actions.hasNext()) {
+                    stream.write((actions.next() + "\n").getBytes(StandardCharsets.UTF_8));
+                }
+            } finally {
+                stream.close();
+            }
+        } else {
+            if (fs.exists(path)) {
+                throw new FileAlreadyExistsException(path.toString());
+            }
+            Path tempPath = createTempPath(path);
+            Boolean streamClosed = false; // This flag is to avoid double close
```

Comment:
nit: this can be `boolean` instead of `Boolean`.

(`LogStore::write` also uses `Boolean` but I should fix that in another PR ...)

Thanks!

Reply:
Will change this